sbin/build.sh

```diff
@@ -606,6 +606,76 @@ createOpenJDKFailureLogsArchive() {
     createArchive "${adoptLogArchiveDir}" "${makeFailureLogsName}"
 }
 
+# Build the CycloneDX Java library and app used for SBoM generation
+buildCyclonedxLib() {
+  local javaHome=""
+
+  if [ ${JAVA_HOME+x} ] && [ -d "${JAVA_HOME}" ]; then
+    javaHome=${JAVA_HOME}
+  elif [ ${JDK8_BOOT_DIR+x} ] && [ -d "${JDK8_BOOT_DIR}" ]; then
+    javaHome=${JDK8_BOOT_DIR}
+  elif [ ${JDK11_BOOT_DIR+x} ] && [ -d "${JDK11_BOOT_DIR}" ]; then
+    javaHome=${JDK11_BOOT_DIR}
+  else
+    echo "Unable to find a suitable JAVA_HOME to build the cyclonedx-lib"
+    exit 2
+  fi
+
+  # We need the exitcode from ant
+  set +eu
+
+  ant -f "${WORKSPACE}/cyclonedx-lib/build.xml" -Djava.home="${javaHome}" clean
+  ant -f "${WORKSPACE}/cyclonedx-lib/build.xml" -Djava.home="${javaHome}" build
+  exitCode=$?
+
+  set -eu
+
+  if [ "${exitCode}" -ne 0 ]; then
+    echo "Failed to build the cyclonedx-lib, exiting"
+    exit ${exitCode}
+  fi
+}
+
+# Generate the SBoM
+generateSBoM() {
+  local javaHome=""
+
+  if [ ${JAVA_HOME+x} ] && [ -d "${JAVA_HOME}" ]; then
+    javaHome=${JAVA_HOME}
+  elif [ ${JDK8_BOOT_DIR+x} ] && [ -d "${JDK8_BOOT_DIR}" ]; then
+    javaHome=${JDK8_BOOT_DIR}
+  elif [ ${JDK11_BOOT_DIR+x} ] && [ -d "${JDK11_BOOT_DIR}" ]; then
+    javaHome=${JDK11_BOOT_DIR}
+  else
+    echo "Unable to find a suitable JAVA_HOME to run the TemurinGenSBOM app"
+    exit 2
+  fi
+
+  # We need the exitcode from the app 
+  set +eu
+
+  # classpath to run CycloneDX java app TemurinGenSBOM
+  classpath="${WORKSPACE}/cyclonedx-lib/build/jar/temurin-gen-sbom.jar:${WORKSPACE}/cyclonedx-lib/build/jar/cyclonedx-core-java.jar:${WORKSPACE}/cyclonedx-lib/build/jar/jackson-core.jar:${WORKSPACE}/cyclonedx-lib/build/jar/jackson-dataformat-xml.jar:${WORKSPACE}/cyclonedx-lib/build/jar/jackson-databind.jar:${WORKSPACE}/cyclonedx-lib/build/jar/jackson-annotations.jar:${WORKSPACE}/cyclonedx-lib/build/jar/json-schema.jar:${WORKSPACE}/cyclonedx-lib/build/jar/commons-codec.jar:${WORKSPACE}/cyclonedx-lib/build/jar/commons-io.jar:${WORKSPACE}/cyclonedx-lib/build/jar/github-package-url.jar"
+
+  # Run app to generate SBoM
+  # List of commands
+  declare -a cmds=('--create temurin_sbom.json --name "Temurin SBOM" --version "1.2.3" --type "application" --author "Adoptium"'
```

Comment:
I'd be tempted to unroll this loop since the use of an array just makes the flow harder to follow. Since we only have two operations I think it'd be simpler to just run them both, and give separate error messages for each section to make it clear which one failed e.g.
```
"${javaHome}"/bin/java -cp "${classpath}" temurin.sbom.TemurinGenSBOM -create temurin_sbom.json --name "Temurin SBOM" --version "1.2.3" --type "application" --author "Adoptium"
RC=$?
if [ "${RC}" -ne 0 ]; then
  echo Failed to ru TemurinGenSBOM. rc=${RC}, exiting
  exit $RC
fi

"${javaHome}"/bin/java -cp "${classpath}" --add_component temurin_sbom.json --name "openjdk" --version "jdk-19+12"
RC=$?
if [ "${RC}" -ne 0 ]; then
  echo Failed to add componenent temurin_sbom.json - rc=${existCode}, exiting
  exit $RC
fi
```
Also, can we avoid hardcoding `jdk-19+12`? - We should have access to the openjdk version number from within this script.

Reply:
The array of cmds is just an example...made up!
The TemurinGenSBOM scaffold just prints it out at the moment
In reality there will be probably be a lot of "operations" to generate SBOM components/dependencies/hashes/etc... hence lots of calls to the SBOM gen, hence I thought array of cmds better, but design TBD... @SehrishHussain 
i'll remove for now

Reply:
changed, does look neater without loop